core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -263,6 +263,9 @@ static void init(String[] args) throws Throwable {
         INSTANCE = new Bootstrap();
 
         boolean foreground = !"false".equals(System.getProperty("es.foreground", System.getProperty("es-foreground")));
+        // Clear these properties because we are done with them. No need to let them leak into the Settings.
+        System.clearProperty("es.foreground");
```

Comment:
I think changing any properties we use to just pass info from the launch script to ES should just be under a different prefix.